src/System.Diagnostics.Metrics.Temp/InstrumentT.cs

```diff
@@ -96,7 +126,7 @@ protected void RecordMeasurement(T measurement)
             T measurement,
             ReadOnlySpan<KeyValuePair<string, object?>> tags)
         {
-            foreach (var listener in this.Listeners)
+            foreach (var listener in this.CachedListeners)
```

Comment:
We could use a for loop now that `CachedListeners` is an array.

Reply:
Agreed. But...

The System.Diagnostics.Metrics.Temp will be replaced with the official .NET API soon.